Ensure future compatibility, drop old code (v1/v2 support)

kolyshkin opened the pull request:
This PR does two things.

1. Make sure that once (if ever) Linux kernel will start supporting capabilities v4, this library won't stop working. This relies on the assumption that v3 will still be supported.

2. Drops v1 and v2 API.

v3 API is used since the Linux kernel 2.6.26.

Since go 1.18 (no longer supported as of go 1.20 release), the minimum Linux kernel requirement is 2.6.32 (see [1]). So, it does not make sense to support capabilities v1 and v2 any more.

Drop the support.

[1] https://tip.golang.org/doc/go1.18#linux

avagin commented on capability/capability_linux.go:
```diff
@@ -100,12 +98,7 @@ func mkString(c Capabilities, max CapType) (ret string) {
-		p.hdr.pid = int32(pid)
-		c = p
-	case linuxCapVer2, linuxCapVer3:
+	case linuxCapVer3:
```
If capVers is unknown version, we need to fall back to linuxCapVer3. Otherwise everything will be broken when a new version is introduced.

kolyshkin replied:
👍🏻 Makes sense, please see updated commits.

kolyshkin committed:
Fix future version compatibility

In case kernel folks will ever release capability v4, the chances are
high v3 is still supported. Therefore, we should not error out upon
seeing an unknown version from the kernel, but assume we can go with v3.

While at it, treat the uninitialized capVers as an error. Before this
patch, it was still treated as an error, but "unknown capability version"
is not exactly what the error is, so let's be more specific.

Reported-by: Andrei Vagin <avagin@google.com>
Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

kolyshkin committed:
Drop capability v1 and v2 support

Capabilities v3 API was added by the Linux kernel 2.6.26.

Since go 1.18 (no longer supported as of go 1.20 release), the minimum
Linux kernel requirement is 2.6.32 (see [1]). So, it does not make sense
to support capabilities v1 and v2 any more.

Drop the support, returning the appropriate error message.

[1] https://tip.golang.org/doc/go1.18#linux

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

avagin commented:
LGTM

kolyshkin commented:
I'm going to maintain a fork at https://github.com/kolyshkin/capability.

kolyshkin commented:
Previously LGTMed by @avagin [here](https://github.com/syndtr/gocapability/pull/23#issuecomment-1458872809).
